src/kernel/arch/aarch64/interrupts.zig

```diff
@@ -28,6 +28,108 @@ comptime {
     );
 }
 
+comptime {
+    asm (
+        \\.globl exception_table
+        \\.balign 0x800
+        \\exception_table:
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+    );
+}
+
+export fn exceptionHandler() noreturn {
+    const exception_entry_number = @truncate(u32, lr() >> 7 & 0xf);
+    var elr_elx: u64 = undefined;
+    var esr_elx: u64 = undefined;
+    var far_elx: u64 = undefined;
+    var sctlr_elx: u64 = undefined;
+    var spsr_elx: u64 = undefined;
+    var vbar_elx: u64 = undefined;
+    if (currentEl() == 3) {
+        elr_elx = mrsEl("elr_el", 3);
+        esr_elx = mrsEl("esr_el", 3);
+        far_elx = mrsEl("far_el", 3);
+        sctlr_elx = mrsEl("sctlr_el", 3);
+        spsr_elx = mrsEl("spsr_el", 3);
+        vbar_elx = mrsEl("vbar_el", 3);
+    } else if (currentEl() == 2) {
+        elr_elx = mrsEl("elr_el", 2);
+        esr_elx = mrsEl("esr_el", 2);
+        far_elx = mrsEl("far_el", 2);
+        sctlr_elx = mrsEl("sctlr_el", 2);
+        spsr_elx = mrsEl("spsr_el", 2);
+        vbar_elx = mrsEl("vbar_el", 2);
+    } else if (currentEl() == 1) {
+        elr_elx = mrsEl("elr_el", 1);
+        esr_elx = mrsEl("esr_el", 1);
+        far_elx = mrsEl("far_el", 1);
+        sctlr_elx = mrsEl("sctlr_el", 1);
+        spsr_elx = mrsEl("spsr_el", 1);
+        vbar_elx = mrsEl("vbar_el", 1);
+    }
```

Comment:
This could be simplified to an inline for loop over the exception levels.

Reply:
At the time, I was trying to get as much comptime as possible, and I think I found a problem with comptime loops. I will revisit that.

Reply:
Ah I see. I gave it a try just now and it seemed to work.

Reply:
You are right. I have changed it to an inline for.